### DISCUSSION TITLE
Ways to monitor connection to Azure IoT Hub/other IoT services

### DISCUSSION THREAD

**Discussion by mikegpl** [Azure/azure-iot-sdk-java#1436] [Q&A] [answered]
Hi,

I'm wondering, are there any guidelines for monitoring the connection to the following services:
- IoT Hub,
- DPS,
- Blob Storage,
- IoT Central API?

In my solution I use a HTTP Proxy which has only a few hostnames (used in communication with Azure) allowed, however, sometimes I get random/transient errors when communicating with Azure. To make sure they are not related with the Proxy setup I would like to set up a some sort of health check which would tell me if it's the one failing is the proxy or the Azure. For security reasons, I don't want to use any of my existing Azure subscriptions for this task - maybe there are some public endpoints that I could check periodically? 

Best regards,
Michał

**Comment by timtay-microsoft** [comment, marked as answer]
Hi there, Michal,

I'm not aware of any such public endpoint that you could ping to check the health of your particular Azure resource. My understanding is that you would need to make a simple request to the service in question in order to get that health check.

You may consider opening a discussion on [this user voice page](https://feedback.azure.com/d365community/forum/fcb810f7-f824-ec11-b6e6-000d3a4f0da0) in order to get some traction with a service side engineer. This discussion board is mainly for discussing the SDK itself, and not the service.

Thanks,
-Tim

**Comment by timtay-microsoft** [comment]
@mikegpl is there anything else I can help you with on this topic? If not, I'll go ahead and close this thread

**Reply by mikegpl** [reply]
Hi @timtay-microsoft, thank you for your response - I'll try opening discussion on the page you mentioned and I'm closing this one.
